### PR TITLE
Beginning of HappyBase batch module.

### PR DESCRIPTION
**Pull request by dhermes**
For comparison: https://github.com/wbolster/happybase/blob/9cbd718c10a3089f234f1eac1236b631e1f8e7cd/happybase/batch.py

### CODE REVIEW

**Comment by jonparrott** [comment]
LGTM with minor readability nit.

(as an aside, I really hate `unittest2`'s testing style)

File: gcloud/bigtable/happybase/batch.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                raise TypeError('When batch_size is set, a Batch cannot be '
+                                'transactional')
+            if batch_size <= 0:
+                raise ValueError('batch_size must be positive')
+
+        self._table = table
+        self._batch_size = batch_size
+        # Timestamp is in milliseconds, convert to microseconds.
+        self._timestamp = self._delete_range = None
+        if timestamp is not None:
```
Comment (jonparrott) [inline gcloud/bigtable/happybase/batch.py]: nit: I typically like to give assignments and blocks some breathing room. I would prefer:

```python
self._timestamp = self._delete_range = None

# Timestamp is in milliseconds, convert to microseconds.
if timestamp is not None:
    ...

self._transaction = transaction

**Comment by dhermes** [comment]
@jonparrott `unittest2` is just a back-port of `unittest`, so there is no testing style to speak of. We do things guided by [some notes][1] from @tseaver.

[1]: https://palladion.com//home/tseaver/obzervationz/2008/unit_testing_notes-20080724

File: gcloud/bigtable/happybase/batch.py (hunk 1 above)
Comment (dhermes) [inline gcloud/bigtable/happybase/batch.py]: FYI: Will fold this into original commit and merge once Travis is green.

**Pull request merged**